src/shared_utilities/implicit_timestepping.jl

```diff
@@ -0,0 +1,166 @@
+using ClimaCore.MatrixFields
+import ClimaCore.MatrixFields: @name, ⋅
+import UnrolledUtilities
+import LinearAlgebra
+import LinearAlgebra: I
+
+export make_tendency_jacobian,
+    make_update_jacobian, set_dfluxBCdY!, ImplicitEquationJacobian
+
+
+"""
+   make_tendency_jacobian(model::AbstractModel)
+
+Creates and returns a function which updates the auxiliary
+variables `p` in place and then updates the entries of the
+Jacobian matrix `W` for the `model` in place.
+
+The default is that no updates are required, no implicit tendency is
+present, and hence the timestepping is entirely explicit.
+
+Note that the returned function `tendency_jacobian!` should be
+used as `Wfact!` in `ClimaTimeSteppers.jl` and `SciMLBase.jl`.
+"""
+function make_tendency_jacobian(model::AbstractModel)
```

Comment:
Maybe we can open an issue to unify the terminology between `make_update_jacobian` and `make_compute_imp_tendency`, and between `make_tendency_jacobian` and `make_imp_tendency`? I think they serve the same roles but it might be clearer if we name them the same. What do you think?

Reply:
(feel free to assign some of these issues to me!)

Reply:
Good point! I was getting confused between these while working on this PR....

Reply:
I opened an issue here https://github.com/CliMA/ClimaLand.jl/issues/660